internal/controller/server/repository/job/job.go

```diff
@@ -61,23 +63,26 @@ func (j jobRepository) Save(ctx context.Context, jobID uint64, executionData *cl
 // Delete function delete the job of given key from pendingList in database
 func (j jobRepository) Delete(ctx context.Context, key string) error {
 	_, err := j.etcdClient.DeleteKey(ctx, constant.JobPendingPrefix+key)
-	return err
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+	return nil
 }
 
 // FetchNextJob returns jobID and executionData from pendingList
 func (j jobRepository) FetchNextJob(ctx context.Context) (string, *clientCPproto.RequestForExecute, error) {
 	keys, values, err := j.etcdClient.GetAllKeyAndValues(ctx, constant.JobPendingPrefix)
 	if err != nil {
-		return "", nil, err
+		return "", nil, status.Error(codes.Internal, err.Error())
 	}
 	if len(values) == 0 {
-		return "", nil, errors.New("no pending job in pending job list")
+		return "", nil, status.Error(codes.NotFound, errors.New(constant.Controller+"no pending job in pending job list").Error())
```

Comment:
i guess in place of errors.New we can just use fmt.Sprintf() or simple string concatenation to join the strings.